fastnode2vec/node2vec.py

```diff
@@ -14,38 +16,77 @@ def set_seed(seed):
 class Node2Vec(Word2Vec):
     def __init__(
         self,
-        graph,
-        dim,
-        walk_length,
-        context,
-        p=1.0,
-        q=1.0,
-        workers=1,
-        batch_walks=None,
-        seed=None,
-        **args,
+        graph: Graph,
+        dim: int,
+        walk_length: int,
+        window: int,
+        p: float = 1.0,
+        q: float = 1.0,
+        workers: Union[int, str] = "auto",
+        batch_walks: Optional[int] = None,
+        use_skipgram: bool = True,
+        seed: Optional[int] = None,
+        **kwargs,
     ):
+        """Create a new instance of Node2Vec.
+
+        Parameters
+        ---------------------------
+        graph: Graph
+            Graph object whose nodes are to be embedded.
+        dim: int
+            The dimensionality of the embedding
+        walk_length: int
+            Length of the random walk to be computed
+        window: int
+            The dimension of the window.
+            This is HALF of the context, as gensim will
+            consider as context all nodes before `window`
+            and after `window`.
+        p: float = 1.0
+            The higher the value, the lower the probability to return to
+            the previous node during a walk.
+        q: float = 1.0
+            The higher the value, the lower the probability to return to
+            a node connected to a previous node during a walk.
+        workers: Union[int str] = "auto"
+            Number of workers to use in GenSim. By default, "auto",
+            that is we use all of the available threads.
+        batch_walks: Optional[int] = None
+            Target size (in words) for batches of examples passed to worker threads (and
+            thus cython routines).(Larger batches will be passed if individual
+            texts are longer than 10000 words, but the standard cython code truncates to that maximum.)
+        use_skipgram: bool = True
```

Comment:
does it make sense to use CBOW? I don't think the original paper uses it. Also it would probably break some of the math (predicting a node from a walk only makes less sense IMO).

Reply:
I have implemented in Rust language versions using CBOW, SkipGram and GloVe that use random walk samples, and in all of the three, there is definitely a signal. It depends on the task, and it can get quite variable, though I agree that SkipGram often performs better.

CBOW also has the perk of being faster than SkipGram of a `window size` factor, at least in gensim's implementation.

Reply:
I really don't think other methods should be used, except by expert users who can modify the code.

Reply:
As you prefer! I'll update these things shortly.

Reply:
Thanks! Just being explicit about this parameter not being standard AT ALL works too.